java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -164,18 +166,40 @@ public void registerSSHMinion(String minionId, Optional<Long> proxyId,
     private void registerMinion(String minionId, boolean isSaltSSH,
                                 Optional<Long> saltSSHProxyId,
                                 Optional<String> activationKeyOverride) {
+
         // Match minions via their machine id
-        Optional<String> optMachineId = SALT_SERVICE.getMachineId(minionId);
-        if (!optMachineId.isPresent()) {
-            LOG.info("Cannot find machine id for minion: " + minionId);
-            return;
-        }
-        //FIXME: refactor this whole function so we don't have to call get
-        //in so many places.
-        String machineId = optMachineId.get();
-        MinionServer minionServer;
-        Optional<User> creator = MinionPendingRegistrationService.getCreator(minionId);
+        Opt.fold(SALT_SERVICE.getMachineId(minionId),
+            () -> {
+                LOG.info("Cannot find machine id for minion: " + minionId);
+                return false;
+            },
+            machineId -> {
+                Optional<User> creator = MinionPendingRegistrationService.getCreator(minionId);
+                if (checkIfMinionAlreadyRegistered(minionId, machineId, creator, isSaltSSH)) {
+                    return true;
+                }
+                if (duplicateMinionNamePresent(minionId)) {
+                    return false;
+                }
+                finalizeMinionRegistration(minionId, machineId, creator,
```

Comment:
Line length is 120 columns

Reply:
which line?

Reply:
you mean the whole statement, although it has a line break in the middle? I am bit confused

Reply:
I only mean we should not add newlines unless we are really breaking the 120 column limit. Here the line is truncated after 73 chars - can we move all parameters on the same line?

Reply:
On this particular line, with current indentation, moving all paramaters into the same line would result in 124 columns (above limit).